classes/01class/exercises/c01-aws01/zarajoy/c01-aws01.md

```diff
@@ -0,0 +1,398 @@
+            "RootDeviceType": "ebs",
+            "SecurityGroups": [
+                {
+                    "GroupName": "my-sg",
+                    "GroupId": "sg-0aa6a164d5faaf958"
+                }
+            ],
+            "SourceDestCheck": true,
+            "StateReason": {
+                "Code": "pending",
+                "Message": "pending"
+            },
+            "VirtualizationType": "hvm",
+            "CpuOptions": {
+                "CoreCount": 1,
+                "ThreadsPerCore": 1
+            },
+            "CapacityReservationSpecification": {
+                "CapacityReservationPreference": "open"
+            },
+            "MetadataOptions": {
+                "State": "pending",
+                "HttpTokens": "optional",
+                "HttpPutResponseHopLimit": 1,
+                "HttpEndpoint": "enabled"
+            }
+        }
+    ],
+    "OwnerId": "512742231244",
+    "ReservationId": "r-07cc8f6e71d7afe6e"
+}
+
+
+- Commands to connect to the first EC2 instance:
+
+z@bacon:~$ ssh -i awskp01.pem ec2-user@13.239.28.37
+Warning: Identity file awskp01.pem not accessible: No such file or directory.
+The authenticity of host '13.239.28.37 (13.239.28.37)' can't be established.
+ECDSA key fingerprint is SHA256:ewimNRcdoLHVsDfJ4iebEcWD920WprRuJxRiQOZkphw.
+Are you sure you want to continue connecting (yes/no/[fingerprint])? yes
+Warning: Permanently added '13.239.28.37' (ECDSA) to the list of known hosts.
+ec2-user@13.239.28.37: Permission denied (publickey,gssapi-keyex,gssapi-with-mic).
+z@bacon:~$ aws ec2 describe-instances --query "Reservations[].Instances[].[InstanceId, PublicIpAddress]"
+[
+    [
+        "i-07b1ba8023f12e889",
+        null
+    ],
+    [
+        "i-00cb16d60056e7353",
+        "13.239.28.37"
+    ]
+]
+z@bacon:~$ ssh -i awskp01.pem ec2-user@13.239.28.37
+Warning: Identity file awskp01.pem not accessible: No such file or directory.
+z@bacon:~$ sudo chmod 400 aws01kp.pem
+[sudo] password for z: 
+z@bacon:~$ ssh -i aws01kp.pem ec2-user@13.239.28.37
+
+       __|  __|_  )
+       _|  (     /   Amazon Linux 2 AMI
+      ___|\___|___|
+
+https://aws.amazon.com/amazon-linux-2/
+4 package(s) needed for security, out of 8 available
+Run "sudo yum update" to apply all updates.
+[ec2-user@ip-172-31-36-157 ~]$ 
+
+
+
+- Create another EC2, also using the CLI instance in the same VPC but with private address only and connect to it.
+- Commands to create the second EC2 instance and any additional resource required:
+z@bacon:~$ aws ec2 create-key-pair --key-name paws01kp --query 'KeyMaterial' --output text > paws01kp.pem
+z@bacon:~$ sudo chmod 400 paws01kp.pem
+[sudo] password for z: 
+z@bacon:~$ aws ec2 create-security-group --group-name SecureGroup --description  "Secure group for DOA"
+{
+    "GroupId": "sg-09c8bcfe909954f1a"
+}
+z@bacon:~$ aws ec2 authorize-security-group-ingress --group-name SecureGroup --protocol tcp --port 22 --cidr 180.150.38.8/32
```

Comment:
can you confirm that the security group for the instance without public IP only have this rule attached to it?

Reply:
z@bacon:~$ aws ec2 describe-instances --instance-ids i-08b574c6436ac7ad8
{
    "Reservations": [
        {
            "Groups": [],
            "Instances": [
                {
                    "AmiLaunchIndex": 0,
                    "ImageId": "ami-088ff0e3bde7b3fdf",
                    "InstanceId": "i-08b574c6436ac7ad8",
                    "InstanceType": "t2.micro",
                    "KeyName": "paws01kp",
                    "LaunchTime": "2020-06-16T06:43:58.000Z",
                    "Monitoring": {
                        "State": "disabled"
                    },
                    "Placement": {
                        "AvailabilityZone": "ap-southeast-2a",
                        "GroupName": "",
                        "Tenancy": "default"
                    },
                    "PrivateDnsName": "ip-172-31-45-4.ap-southeast-2.compute.internal",
                    "PrivateIpAddress": "172.31.45.4",
                    "ProductCodes": [],
                    "PublicDnsName": "",
                    "State": {
                        "Code": 16,
                        "Name": "running"
                    },
                    "StateTransitionReason": "",
                    "SubnetId": "subnet-48638000",
                    "VpcId": "vpc-ce545da9",
                    "Architecture": "x86_64",
                    "BlockDeviceMappings": [
                        {
                            "DeviceName": "/dev/xvda",
                            "Ebs": {
                                "AttachTime": "2020-06-16T06:43:59.000Z",
                                "DeleteOnTermination": true,
                                "Status": "attached",
                                "VolumeId": "vol-0338c6387c6e221c4"
                            }
                        }
                    ],
                    "ClientToken": "",
                    "EbsOptimized": false,
                    "EnaSupport": true,
                    "Hypervisor": "xen",
                    "NetworkInterfaces": [
                        {
                            "Attachment": {
                                "AttachTime": "2020-06-16T06:43:58.000Z",
                                "AttachmentId": "eni-attach-0f2eb3e32c4e8fde4",
                                "DeleteOnTermination": true,
                                "DeviceIndex": 0,
                                "Status": "attached"
                            },
                            "Description": "",
                            "Groups": [
                                {
                                    "GroupName": "SecureGroup",
                                    "GroupId": "sg-09c8bcfe909954f1a"
                                }
                            ],
                            "Ipv6Addresses": [],
                            "MacAddress": "06:87:b3:bd:c3:20",
                            "NetworkInterfaceId": "eni-01e2e4d50fd097c6b",
                            "OwnerId": "512742231244",
                            "PrivateDnsName": "ip-172-31-45-4.ap-southeast-2.compute.internal",
                            "PrivateIpAddress": "172.31.45.4",
                            "PrivateIpAddresses": [
                                {
                                    "Primary": true,
                                    "PrivateDnsName": "ip-172-31-45-4.ap-southeast-2.compute.internal",
                                    "PrivateIpAddress": "172.31.45.4"
                                }
                            ],
                            "SourceDestCheck": true,
                            "Status": "in-use",
                            "SubnetId": "subnet-48638000",
                            "VpcId": "vpc-ce545da9",
                            "InterfaceType": "interface"
                        }
                    ],
                    "RootDeviceName": "/dev/xvda",
                    "RootDeviceType": "ebs",
                    "SecurityGroups": [
                        {
                            "GroupName": "SecureGroup",
                            "GroupId": "sg-09c8bcfe909954f1a"
                        }
                    ],
                    "SourceDestCheck": true,
                    "VirtualizationType": "hvm",
                    "CpuOptions": {
                        "CoreCount": 1,
                        "ThreadsPerCore": 1
                    },
                    "CapacityReservationSpecification": {
                        "CapacityReservationPreference": "open"
                    },
                    "HibernationOptions": {
                        "Configured": false
                    },
                    "MetadataOptions": {
                        "State": "applied",
                        "HttpTokens": "optional",
                        "HttpPutResponseHopLimit": 1,
                        "HttpEndpoint": "enabled"
                    }
                }
            ],
            "OwnerId": "512742231244",
            "ReservationId": "r-05d1863c7d002fa30"
        }
    ]
}

Reply:
cant work out a command to just show instanceID and groups associated :/